libnymea-gpio/gpiobutton.cpp

```diff
@@ -197,16 +197,14 @@ bool GpioButton::enable()
     disable();
 
     m_monitor = new GpioMonitor(m_gpioNumber, this);
-    m_monitor->setEdge(Gpio::EdgeBoth);
-    m_monitor->setActiveLow(m_activeLow);
 
-    if (!m_monitor->enable()) {
+    if (!m_monitor->enable(m_activeLow, Gpio::EdgeBoth)) {
         qCWarning(dcGpio()) << "Could not enable GPIO monitor for" << this;
         delete m_monitor;
         m_monitor = nullptr;
         return false;
     }
-    connect(m_monitor, &GpioMonitor::interruptOccurred, this, &GpioButton::onInterruptOccurred);
+    connect(m_monitor, &GpioMonitor::valueChanged, this, &GpioButton::onValueChanged);
```

Comment:
still not happy with `valueChanged`, if you monitor for lets say only rising edges, the actual value does not change...it is just an interrupt saying: now it happened again. 
We can keep it for now since there was too much confusion in the first place :-D just sharing the thoughts behind this

Reply:
Yes... I get your point.. Interruptoccurred is IMO misleading in the same way though. For a start. technically it isn't an interrupt and for the regular use case of monitoring an input it's confusing enough so that you used it wrong yourself in the plugin by connecting to "enabledChanged" instead of "interruptOcurred" accidentally to read the value. That most likely wouldn't have happened if a valueChanged was there.

Then, I suppose, given you can configure the monitor to only react on rising/falling flags, one *could* argue that valueChanged still works, after all it did change, you just configured to be notified only on 0 or 1.

anyhow, the main reason why it's valueChanged again is as I just wasn't able to fix the new code in numerous tries and reverted to the old known-to-work code without touching it at all any more.

That said, in the meantime I've read through pythons RPi.GPIO lib (which is written in C) and noticed that they actually hook into /dev/mem via memory mapping. At a first glance it looked like it's still being polled in a loop, which I honestly hate to do (poll threads are the worst, regardless if they poll memory or sysfs files) but at least that seemed to give control flexibility over literally everything, like individually configuring the pull resitors and all. That may be way to go once someone dares to touch this again.

Reply:
using kernel memory directly is super bad, and poll threads are the worst, I agree. 
I think the rpi gpio stuff is not a good reference. libgpiod is the way to go once we touch this again.